archive/progress_common/src/main/scala/uk/ac/wellcome/platform/archive/common/progress/monitor/ProgressTracker.scala

```diff
@@ -84,27 +86,33 @@ class ProgressTracker @Inject()(
       .given(attributeExists('id))
       .update('id -> update.id, mergedUpdate)
 
-    Scanamo.exec(dynamoClient)(ops) match {
+    Scanamo.exec(dynamoDbClient)(ops) match {
       case Left(ConditionNotMet(e: ConditionalCheckFailedException)) => {
         val idConstraintError =
           IdConstraintError(s"Progress does not exist for id:${update.id}", e)
 
         Failure(idConstraintError)
       }
-
       case Left(scanamoError) => {
         val exception = new RuntimeException(scanamoError.toString)
         warn(s"Failed to update Dynamo record: ${update.id}", exception)
         Failure(exception)
       }
-
-      case r @ Right(progress) => {
+      case Right(progress) => {
         debug(
           s"Successfully updated Dynamo record: ${update.id}, got $progress")
         Success(progress)
       }
     }
   }
+
+  def findByBagId(bagId: BagId) = {
```

Comment:
Does this return `List[BagIngest]`? Or a Future thereof?

Reply:
A list, `get` returns a Future, do you think it should be the same?